### DISCUSSION TITLE
The signed distance from a given point to a 2D slice of a given 3D mesh (for example, obj file)

### DISCUSSION THREAD

**Discussion by Pupillen** [libigl/libigl#2221] [Q&A]
Given a 3D mesh file (.obj) with a point, make a slice across the point to intersect this 3D mesh to get a 2D slice. How to get the signed distance of this point to this 2D profile efficiently? Is there a relevant algorithm or graphics library I can refer to? Thanks a lot！

**Comment by Pupillen** [comment]
And If the cutting plane and the corresponding 2D slice are not fixed but change very often (1000HZ), how can I improve efficiency? I want a very high-performance program.

**Comment by alecjacobson** [comment]
I'd recommend trying to do something specific for slice planes. If you want a very exact result you could use https://github.com/libigl/libigl/blob/main/include/igl/copyleft/cgal/intersect_with_half_space.h

This problem sounds similar to slicing for 3D printing. In that case the slice plan can be chosen to be one of the canonical axes (e.g., the z axis). Then you could intersect each triangle to get 0 or more edges.
